src/OpenSearch/LegacyTransportWrapper.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace OpenSearch;
+
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * Transport that wraps the legacy transport.
+ *
+ * @deprecated in 2.3.2 and will be removed in 3.0.0. Use PsrTransport instead.
+ */
+class LegacyTransportWrapper implements TransportInterface
+{
+    public function __construct(
+        protected Transport $transport,
+    ) {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function sendRequest(
+        string $method,
+        string $uri,
+        array $params = [],
+        mixed $body = null,
+        array $headers = [],
+    ): array|string|null {
+        $promise = $this->transport->performRequest($method, $uri, $params, $body);
+        return $this->transport->resultOrFuture($promise);
```

Comment:
`resultOrFuture` will return a FutureArray of ringphp in case of sync mode (which is the case here). It does not actually return an array. This will fail with the native return type which is enforced by PHP. You need to extract the actual data (using `$futureArray->_value` to read it)